pkg/proxy/ipvs/proxier.go

```diff
@@ -525,6 +534,31 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 	return append(mods, bmods...), nil
 }
 
+// GetKernelVersion returns currently running kernel version.
+func (handle *LinuxKernelHandler) GetKernelVersion() (string, error) {
+	kernelVersionFile := "/proc/sys/kernel/osrelease"
+	f, err := os.Open(kernelVersionFile)
+	if err != nil {
+		return "", fmt.Errorf("error opening osrelease file %q: %v", kernelVersionFile, err)
+	}
+	defer f.Close()
+
+	bf := bufio.NewScanner(f)
```

Comment:
Curious why not [ioutil.ReadFile](https://golang.org/src/io/ioutil/ioutil.go?s=1503:1549#L42)?

Reply:
`ReadFile` reads all to memory and the issue you linked suggest 1K limit if i remember correctly.

Reply:
hmm good point, I think `iotuil.ReadFile` is OK given the assumptions around `/proc/sys/kernel/osrelease`. If we want the 1KB limit, I don't think a Scanner is required can just use a bytes.Buffer (see [bytes.NewBuffer](/proc/sys/kernel/osrelease)) and read from `os.File` returned from `os.Open`.